Actions/AnalyzeTests/TestResultAnalyzer.ps1

```diff
@@ -1,107 +1,324 @@
-        foreach($appName in $appNames) {
-            $appTests = 0
-            $appTime = 0.0
-            $appFailed = 0
-            $appSkipped = 0
-            $suites = $testResults.testsuites.testsuite | where-Object { $_.Properties.property | Where-Object { $_.Value -eq $appName } }
-            foreach($suite in $suites) {
-                $appTests += [int]$suite.tests
-                $appFailed += [int]$suite.failures
-                $appSkipped += [int]$suite.skipped
-                $appTime += [decimal]::Parse($suite.time, [System.Globalization.CultureInfo]::InvariantCulture)
-            }
-            $appPassed = $appTests-$appFailed-$appSkipped
-            Write-Host "- $appName, $appTests tests, $appPassed passed, $appFailed failed, $appSkipped skipped, $appTime seconds"
-            $summarySb.Append("|$appName|$appTests|") | Out-Null
-            if ($appPassed -gt 0) {
-                $summarySb.Append("$($appPassed) :white_check_mark:") | Out-Null
-            }
-            $summarySb.Append("|") | Out-Null
-            if ($appFailed -gt 0) {
-                $summarySb.Append("$($appFailed) :x:") | Out-Null
+    if (Test-Path -Path $testResultsFile -PathType Leaf) {
+        $testResults = [xml](Get-Content -path $testResultsFile -Encoding UTF8)
+        $totalTests = 0
+        $totalTime = 0.0
+        $totalFailed = 0
+        $totalSkipped = 0
+        if ($testResults.testsuites) {
+            $appNames = @($testResults.testsuites.testsuite | ForEach-Object { $_.Properties.property | Where-Object { $_.Name -eq "appName" } | ForEach-Object { $_.Value } } | Select-Object -Unique)
+            if (-not $appNames) {
+                $appNames = @($testResults.testsuites.testsuite | ForEach-Object { $_.Properties.property | Where-Object { $_.Name -eq "extensionId" } | ForEach-Object { $_.Value } } | Select-Object -Unique)
             }
-            $summarySb.Append("|") | Out-Null
-            if ($appSkipped -gt 0) {
-                $summarySb.Append("$($appSkipped) :white_circle:") | Out-Null
+            foreach($testsuite in $testResults.testsuites.testsuite) {
+                $totalTests += $testsuite.Tests
+                $totalTime += [decimal]::Parse($testsuite.time, [System.Globalization.CultureInfo]::InvariantCulture)
+                $totalFailed += $testsuite.failures
+                $totalSkipped += $testsuite.skipped
             }
-            $summarySb.Append("|$($appTime)s|\n") | Out-Null
-            if ($appFailed -gt 0) {
-                $failuresSb.Append("<details><summary><i>$appName, $appTests tests, $appPassed passed, $appFailed failed, $appSkipped skipped, $appTime seconds</i></summary>\n") | Out-Null
+            Write-Host "$($appNames.Count) TestApps, $totalTests tests, $totalFailed failed, $totalSkipped skipped, $totalTime seconds"
+            $summarySb.Append('|Test app|Tests|Passed|Failed|Skipped|Time|\n|:---|---:|---:|---:|---:|---:|\n') | Out-Null
```

Comment:
Markdown generation seems too embedded in the logic. Consider separating it by first constructing the data (e.g. array of objects that holds info about the app, passed tests, etc) and then generating the markdown from that object.

E.g.:
```
$testResultSummary = GetTestResults ...
# @( @{ appName = 'app1', totalTests = 10, passedTests = 8,... }, ... )

# construct MD
$testResultSummaryMD = GetTestResultSummaryMD $testResultSummary
```

Reply:
The test result data is collected first - what happens during markdown generation is grabbing the data which needs to be in that field.
It was done like this to make it easier to modify the content of the markdown if we decided to add different numbers or aggregate differently or include more lines.

Reply:
I don't see how it will be easier to modify the content of the markdown. If we want to, say, add another column `Timed out` (for tests that timed out) or re-order the existing columns, the changes in this script will be too much and too error prone.

How about using a template with placeholders?

Reply:
The comments here are in the section creating MD for normal test results.
This mechanism is actually unchanged in this PR.
Line 306 is where BCPT Result lines are added.
Yes, I am sure it could be refactored to be smarter and better - but I won't be able to pay this tax now.